src/components/container/ConnectGarmin/ConnectGarmin.stories.tsx

```diff
@@ -38,4 +38,16 @@ export const NotEnabledHide = Template.bind({});
 NotEnabledHide.args = { previewState: "notEnabled", title: "Garmin", disabledBehavior: "hide" };
 
 export const NotEnabledDisplayError = Template.bind({});
-NotEnabledDisplayError.args = { previewState: "notEnabled", title: "Garmin", disabledBehavior: "displayError" };
\ No newline at end of file
+NotEnabledDisplayError.args = { previewState: "notEnabled", title: "Garmin", disabledBehavior: "displayError" };
+
+export const HideConnectedNotConnected = Template.bind({});
+HideConnectedNotConnected.args = { previewState: "notConnected", title: "Garmin", hideWhenConnected: true };
+
+export const HideConnectedUnauthorized = Template.bind({});
+HideConnectedUnauthorized.args = { previewState: "unauthorized", title: "Garmin", hideWhenConnected: true };
```

Comment:
This is covered under Unauthorized. It doesn't feel like a special case. I am open to discuss why this should be different

Reply:
This is the "test case" that was failing.  Previously, this would be hidden, despite the connection being in a "needs to reconnect" state.  After the logic change, this is no longer hidden when the "hideWhenConnected" flag is set.  

Now the that logic is changed, the behavior is identical in both Unauthorized cases, and it does seem to clutter up sidebar quite a bit with all of these "test cases" for this flag left in.  I'm not sure the right balance of covering all of the behind the scenes logic with keeping the UI crisp and functional for what the end user will want to see.

Reply:
I think it makes sense to have the extra permutations, even if the outcome is identical to a different setup.  It confirms the new functionality.  If it wasn't here, there would be no real assertion that the new functionality works the way it does.

That said, I was also getting frustrated by having all the permutations written out, too, especially for complex components with many variables (and thus many permutations).  My solutions was to start using a single `Default` case, which just has configurable preview states and settings.  A storybook user can then go in and test any permutations they would like rather than relying on just the hard coded permutations.

Here are a couple examples:

https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/container/InboxItemList/InboxItemList.stories.tsx
https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/asthma/components/AsthmaLogEntryHeader/AsthmaLogEntryHeader.stories.tsx

Reply:
This is a great solution and I have implemented it for the hideWhenConnected uses cases.  There is now only one story per component, with a radio select for the multiple options of previewState.  

In order for this update to be view in real time in storybook, it was needed to add `props.previewState` to the list of variable dependencies of the useEffect function.  Is this the standard way to ensure this functionality works in storybook, or is there a preferred method?